src/test.core.cpp

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+
+// Include all headers.
+#include <openassetio/errors.h>
+#include <openassetio/BatchElementError.hpp>
+#include <openassetio/Context.hpp>
+#include <openassetio/EntityReference.hpp>
+#include <openassetio/InfoDictionary.hpp>
+#include <openassetio/TraitsData.hpp>
+#include <openassetio/hostApi/HostInterface.hpp>
+#include <openassetio/hostApi/Manager.hpp>
+#include <openassetio/hostApi/ManagerFactory.hpp>
+#include <openassetio/hostApi/ManagerImplementationFactoryInterface.hpp>
+#include <openassetio/log/ConsoleLogger.hpp>
+#include <openassetio/log/LoggerInterface.hpp>
+#include <openassetio/log/SeverityFilter.hpp>
+#include <openassetio/managerApi/Host.hpp>
+#include <openassetio/managerApi/HostSession.hpp>
+#include <openassetio/managerApi/ManagerInterface.hpp>
+#include <openassetio/managerApi/ManagerStateBase.hpp>
+#include <openassetio/trait/TraitBase.hpp>
+#include <openassetio/trait/collection.hpp>
+#include <openassetio/trait/property.hpp>
+#include <openassetio/typedefs.hpp>
```

Comment:
Blimey!

All needed for two lines of C++ code?

Reply:
:grin: As mentioned in https://github.com/OpenAssetIO/OpenAssetIO-Test-CMake/pull/1#discussion_r1013846233 this is just used to check that headers exist and can be compiled.

I'll update the code comment to make this clear.

Reply:
Updated code comment in d5fdb87a